### PR TITLE
chore: Remove various unused template tags

### PR DESCRIPTION
**Pull request by larseggert**

### CODE REVIEW

**Comment by rjsparks** [comment]
The title of this PR implies it is simply cleaning away unused cruft, but instead, it makes changes to the output of several templates.

File: ietf/doc/templatetags/ietf_filters.py [hunk 1]
```diff
@@ -142,45 +125,11 @@ def bracketpos(pos,posslug):
...
-    added
-    """
-    string = str(string)
-    if string[:3].lower() == "rfc" and string[3] != " ":
-        return string[:3].upper() + " " + string[3:]
-    else:
-        return string
-
-@register.filter(name='rfcnospace')
-def rfcnospace(string):
```
Comment (rjsparks) [inline ietf/doc/templatetags/ietf_filters.py:159]: The ability to have more than one representation (with and without space) was intentional. While we moved most of the datatracker to having the space, there was concern about disrupting the workflow of external entities using the IPR system. If changing the output wasn't what you meant to do, there's some further suggestion below.

File: ietf/doc/templatetags/ietf_filters.py [hunk 2]
```diff
@@ -142,45 +125,11 @@ def bracketpos(pos,posslug):
...
-    """
-    string = str(string)
-    if string[:3].lower() == "rfc" and string[3] == " ":
-        return string[:3] + string[4:]
-    else:
-        return string
-
 @register.filter
 def prettystdname(string):
     from ietf.doc.utils import prettify_std_name
```
Comment (rjsparks) [inline ietf/doc/templatetags/ietf_filters.py:130]: This function takes an argument named `spacing`, and the filter could be modified to take the argument as well. In the places this filter is used to replace `rfcnospace`, the filter could then be called with `prettystdname:""`

File: ietf/templates/api/index.html [hunk 3]
```diff
@@ -284,7 +284,7 @@ <h2 id="signing-keys">
         <p>
             When sending notifications to other APIs, the datatracker may sign
             information with a
-            <a href="https://datatracker.ietf.org/doc/html/rfc7515">
+            <a href="/doc/html/rfc7515">
```
Comment (rjsparks) [inline ietf/templates/api/index.html:287]: All of these should instead be changed to use the url resolver, e.g. `<a href="{% url 'ietf.doc.views_doc.document_html' name='rfc7515' %}>`

**Review by rjsparks — CHANGES_REQUESTED** [review]

File: ietf/templates/api/index.html (hunk 3 above)
Comment (larseggert) [reply ietf/templates/api/index.html:287]: There are many more that don't use the URL resolver, which is why I made the change in this way - want me to change those as well?

Comment (rjsparks) [reply ietf/templates/api/index.html:287]: Yes, we should move away from having the templates assume they know what the URL structure looks like.

File: ietf/templates/liaisons/field_help.html [hunk 4]
```diff
@@ -34,7 +34,7 @@ <h1>Liaison statement field help</h1>
     <p>
         For definitive information on generating liaison statements, please
         see
-        <a href="{{ "4053"|rfcurl }}">RFC 4053 (BCP 103) "Procedures for Handling Liaison Statements to and from the IETF."</a>
+        <a href="/doc/html/rfc4053">RFC 4053 (BCP 103) "Procedures for Handling Liaison Statements to and from the IETF."</a>
```
Comment (rjsparks) [inline ietf/templates/liaisons/field_help.html:37]: still a bunch of these in the PR?

**Review by rjsparks — CHANGES_REQUESTED** [review]

**Review by rjsparks — APPROVED** [review]

**Pull request merged**